orchestrator/schedules/service.py

```diff
@@ -0,0 +1,279 @@
+    if trigger_name == "cron":
+        return CronTrigger(**trigger_kwargs)
+    if trigger_name == "date":
+        return DateTrigger(**trigger_kwargs)
+    raise ValueError(f"Invalid trigger type: {trigger_name}")
+
+
+def _update_scheduled_task(payload: APSchedulerJobUpdate, scheduler_connection: BaseScheduler) -> None:
+    """Update an existing scheduled task in the scheduler.
+
+    Only allow update of name and trigger
+    Job id must be that of an existing job
+    Do not insert in linker table - it should already exist.
+
+    Args:
+        payload: APSchedulerJobUpdate The scheduled task to update.
+        scheduler_connection: BaseScheduler The scheduler connection.
+    """
+    logger.info(f"Updating scheduled task: {payload}")
+
+    schedule_id = str(payload.schedule_id)
+    job = scheduler_connection.get_job(job_id=schedule_id)
+    if not job:
+        raise ValueError(f"Schedule Job with id {schedule_id} does not exist.")
+
+    trigger = _build_trigger_on_update(payload.trigger, payload.trigger_kwargs or {})
+    modify_kwargs = {}
+
+    if trigger:
+        job = job.reschedule(trigger=trigger)
+
+    if payload.name:
+        modify_kwargs["name"] = payload.name
+
+    job.modify(**modify_kwargs)
+
+
+def _delete_scheduled_task(payload: APSchedulerJobDelete, scheduler_connection: BaseScheduler) -> None:
+    """Delete an existing scheduled task in the scheduler and also in the linker table.
+
+    Args:
+        payload: APSchedulerJobDelete The scheduled task to delete.
+        scheduler_connection: BaseScheduler The scheduler connection.
+    """
+    logger.info(f"Deleting scheduled task: {payload}")
+
+    schedule_id = str(payload.schedule_id)
+    scheduler_connection.remove_job(job_id=schedule_id)
+    _delete_linker_entry(workflow_id=payload.workflow_id, schedule_id=schedule_id)
+
+
+def workflow_scheduler_queue(queue_item: tuple[str, bytes], scheduler_connection: BaseScheduler) -> None:
+    """Process an item from the scheduler queue.
+
+    Args:
+        queue_item: tuple[str, bytes] The item from the scheduler queue.
+        scheduler_connection: BaseScheduler The scheduler connection.
+    """
+    try:
+        _, bytes_dump = queue_item
+        payload = deserialize_payload(bytes_dump)
+
+        if payload._scheduled_type == SCHEDULER_Q_CREATE:
+            _add_scheduled_task(cast(APSchedulerJobCreate, payload), scheduler_connection)
+        elif payload._scheduled_type == SCHEDULER_Q_UPDATE:
+            _update_scheduled_task(cast(APSchedulerJobUpdate, payload), scheduler_connection)
+        elif payload._scheduled_type == SCHEDULER_Q_DELETE:
+            _delete_scheduled_task(cast(APSchedulerJobDelete, payload), scheduler_connection)
+        else:
+            logger.warning(f"Unexpected schedule type: {payload._scheduled_type}")
+    except Exception as e:
+        logger.exception(e)
+        logger.error(f"Error processing scheduler queue item: {e}")
```

Comment:
`logger.exception(f"Error processing scheduler queue item")` is enough as it logs both the error message and the traceback